src/ArrayParameterType.php

```diff
@@ -2,6 +2,10 @@
 
 namespace Doctrine\DBAL;
 
+/**
+ * @psalm-import-type ParameterTypeOfValue from ParameterType
+ * @psalm-type ArrayParameterTypeOfValue self::*|ParameterTypeOfValue
```

Comment:
ArrayParameterTypeOfValue is a bad naming. It should be an array parameter type, but in fact it's a mix of array parameter type or parameter type.

My suggestion:

ParameterTypeOfValue -> SingleParameterTypeOfValue
ArrayParameterTypeOfValue -> SingleOrArrayParameterTypeOfValue

Reply:
It is and we already have types for this on 4.0.x. I'm working on backporting those.